tests/unit_tests/oci/test_instance.py

```diff
@@ -0,0 +1,475 @@
+"""Tests for pycloudlib's OCI Instance class."""
+
+from unittest import mock
+
+import pytest
+import toml
+
+from pycloudlib.config import Config
+from pycloudlib.errors import PycloudlibError
+from pycloudlib.oci.instance import OciInstance
+
+
+@pytest.fixture
+def oci_instance():
+    """
+    Fixture for OciInstance class.
+
+    This fixture mocks the OCI configuration and the ComputeClient and VirtualNetworkClient classes
+    so that a proper OciInstance object can be created and used for each test. The fixture also
+    mocks the pycloudlib.config.parse_config function to return a Config object with the default
+    values from the pycloudlib.toml.template file.
+    """
+    key_pair = mock.Mock()
+    instance_id = "ocid1.instance.oc1..exampleuniqueID"
+    compartment_id = "ocid1.compartment.oc1..exampleuniqueID"
+    availability_domain = "Uocm:PHX-AD-1"
+
+    # Mock OCI configuration
+    oci_config = {
+        "user": "user_ocid",
+        "fingerprint": "fingerprint",
+        "key_file": "path/to/key_file",
+        "tenancy": "tenancy_ocid",
+        "region": "us-phoenix-1",
+    }
+
+    with mock.patch(
+        "pycloudlib.oci.instance.oci.config.from_file",
+        return_value=oci_config,
+    ), mock.patch(
+        "pycloudlib.oci.instance.oci.core.ComputeClient"
+    ) as mock_compute_client_class, mock.patch(
+        "pycloudlib.oci.instance.oci.core.VirtualNetworkClient"
+    ) as mock_network_client_class, mock.patch(
+        "pycloudlib.config.parse_config"
+    ) as mock_parse_config:
+        # mock loading the config file by just reading in the template, which we know will exist
+        mock_parse_config.return_value = toml.load("pycloudlib.toml.template", _dict=Config)
```

Comment:
Would be nice to use pycloudlib_config = tmp_path / "pyproject.toml"; pycloudlib_config.write_text here too.